subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -659,6 +659,19 @@ uint8_t ll_adv_enable(uint8_t enable)
 
 	adv = is_disabled_get(handle);
 	if (!adv) {
+		/* Already enabled can cause the random address to change */
+		if (IS_ENABLED(CONFIG_BT_CTLR_PRIVACY)) {
+			ull_filter_rpa_refresh();
```

Comment:
Shouldn't this (as per spec) also restart duration and events counter?

Reply:
Good catch, will update with a fix.

Reply:
The text says "can" not "shall". If you have a timer for this already you don't have to update the RPAs here.
The host wouldn't rely on something like this for behavior reason, since it cannot know how exactly the controller acts here.

However on the LE Set Extended Advertising Set Enable command it says:
"If the HCI_LE_Set_Extended_Advertising_Enable command is sent again for
an advertising set while that set is enabled, the timer used for the duration and
the number of events counter are reset and any change to the random address
shall take effect."

So any updates by the host for the advertising set random address should be updated at this point if not already.

Reply:
> The text says "can" not "shall". If you have a timer for this already you don't have to update the RPAs here.
The host wouldn't rely on something like this for behavior reason, since it cannot know how exactly the controller acts here.

"can" states a possibility and/or capability. And the possibility/capability to refresh address is a security feature in the implementation. Hence, will keep the RPA refresh if advertising is enabled on a already enabled case.

Reply:
I see this more as allowing the implementation to shorten the RPA timer if that is easier for the implementation. The security requirement is met as long as the timer is working correctly. Might be easier for implementations that always refresh the RPA of the advertiser when enabled (like the SoftDevice).
You don't force an RPA timeout when starting the disabled advertiser, so no reason to force it now IMO. Not a big deal though, but it would force RPA timeout for all users of the RPA timer (i.e all advertising sets and scanner/initiator). Which seems wasteful IMO.

Not a big deal though. Just a simplification you could do.

However applying the random address change should be enforced here.

Reply:
> You don't force an RPA timeout when starting the disabled advertiser, so no reason to force it now.

hmmm, yes, RPAs are only updated every `rpa_timeout_ms`. I will remove the update of all RPAs on enabling on already enabled advertiser.

> However applying the random address change should be enforced here.

Will work on this.

Reply:
> applying the random address change should be enforced here

Done